tests/test_eigen_tensor.inl

```diff
@@ -7,9 +7,7 @@
 
 #include <pybind11/eigen/tensor.h>
 
-#include "pybind11_tests.h"
-
-PYBIND11_NAMESPACE_BEGIN(PYBIND11_TEST_EIGEN_TENSOR_NAMESPACE)
```

Comment:
Could we keep this?

That would seem safer, to avoid accidental name collisions with other `TEST_SUBMODULE`s, for the case of test_eigen_tensor. For the case of test_eigen_tensor_avoid_stl_array it doesn't really matter, but I'd keep it anyway.